krun/vm_defs.py

```diff
@@ -110,10 +110,30 @@ def _run_exec(self, args, heap_lim_k, bench_env_changes=None):
         p = subprocess.Popen(
             actual_args, stdout=subprocess.PIPE, stderr=subprocess.PIPE,
             env=use_env)
-        stdout, stderr = p.communicate()
+
+        stderr_lines = []
+        while True:
+            line = p.stderr.readline()
+
+            if line == "":
+                break  # EOF, child has exited
+
+            line = line.strip()
+            info(line)
+            stderr_lines.append(line)
+
+        stderr = "\n".join(stderr_lines)
+        stdout = p.stdout.read()
+        p.poll()  # sets returncode attr
```

Comment:
I understand you latter point, but the former about switching stuff around, can you explain a bit more? We need the readline() in a loop to that we get the lines as they appear. There's little we can do if stderr closes but the process keeps running. Perhaps we need to replace the poll() with something which will block.